code/game/objects/items/plushes.dm

```diff
@@ -599,6 +599,35 @@
 	icon_state = "goat"
 	desc = "Despite its cuddly appearance and plush nature, it will beat you up all the same. Goats never change."
 	squeak_override = list('sound/weapons/punch1.ogg'=1)
+	/// Whether or not this goat is currently taking in a monsterous doink
+	var/going_hard = FALSE
+
+/obj/item/toy/plush/goatplushie/attackby(obj/item/clothing/mask/cigarette/rollie/fat_dart, mob/user, params)
+	if(!istype(fat_dart))
+		return ..()
+	if(going_hard)
+		to_chat(user, span_notice("[src] is already going too hard!"))
+		return
+	if(!fat_dart.lit)
+		to_chat(user, span_notice("You'll have to light that first!"))
+		return
+	to_chat(user, span_notice("You put [fat_dart] into [src]'s mouth."))
+	qdel(fat_dart)
+	going_hard = TRUE
+	update_icon(UPDATE_OVERLAYS)
+
+
+/obj/item/toy/plush/goatplushie/examine()
+	. = ..()
+	if(!going_hard)
```

Comment:
this doesn't have to be an early return but it's fine that it is, just for future reference. it's fine for something to end indented, it's bad for a trailing
```dm
else
    //one line of code
```

if that makes sense. again also fine that you did this and the `update_overlays()` bit

Reply:
Ohhh alright yea, got it